WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProductVariationToOrderViewModel.swift

```diff
@@ -38,7 +38,15 @@ final class AddProductVariationToOrderViewModel: ObservableObject {
     /// All purchasable product variations for the product.
     ///
     private var productVariations: [ProductVariation] {
-        productVariationsResultsController.fetchedObjects.filter { $0.purchasable }
+        productVariationsResultsController.fetchedObjects
+            .filter { $0.purchasable }
+            .sorted {
+                if $0.menuOrder == $1.menuOrder {
+                    return ProductVariationFormatter().generateName(for: $0, from: productAttributes) <
+                        ProductVariationFormatter().generateName(for: $1, from: productAttributes)
+                }
+                return $0.menuOrder < $1.menuOrder
```

Comment:
Since the "name" is a concept made-up by the app, I wonder if we could just sort them based on its `variation ID` if the menu order is the same. 

That way: 
- If the menu order is the same, they will be sorted by their creation order
- We could sort at the results controller level which is more performant.

What do you think? Also, do we know how core handles this?

Reply:
Sure, I did it this way because for platform parity with Android (which is falling back to the name if menu order isn't set) but I agree with the reasoning for menu order. It looks like core defaults to variation ID (descending) as the sort order unless you explicitly reorder them, so that would be consistent with the core behavior, too. I'll work on that change.

Reply:
Updated in [ebb4f92](https://github.com/woocommerce/woocommerce-ios/pull/6427/commits/ebb4f924fccc52e344cfe965d55d3e8ff39dd7c4). @Ecarrion ready for another look!